libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/MapWayname.java

```diff
@@ -71,12 +71,15 @@ public void updateDefaultMapTopPadding(int topPadding) {
 
   private void updateLayerWithRoadLabelFeatures(List<Feature> roads, SymbolLayer waynameLayer) {
     if (!roads.isEmpty()) {
```

Comment:
IMO `updateLayerWithRoadLabelFeatures` is getting too long.
What about extracting blocks of code into well-named `private` methods and simplifying conditional expressions somehow?